### PR TITLE
Update to remove OSG CA stuff (SW-3143)

### PR DESCRIPTION
**Pull request by sthapa**

### CODE REVIEW

File: docs/security/user-certs.md [hunk 1]
```diff
@@ -10,13 +10,23 @@ Getting a User Certificate
 --------------------------
 
 This section describes how to get and set up a personal certificate to use on OSG.
-Currently you can get a certificate from either the OSG CA or from CILogon.
-After May 31st, you will only be able to get a certificate issued by CILogon.
-Other CAs may be used; if your virtual organization (VO) requires that you get a certificate from a different CA, [contact your VO Support Center](http://www.opensciencegrid.org/?pid=1000187) for instructions.
+You need an user certificate if you are going to interact directly with OSG resources or infrastructure.
```
Comment (brianhlin) [inline docs/security/user-certs.md]: a user*

File: docs/security/user-certs.md [hunk 2]
```diff
@@ -10,13 +10,23 @@ Getting a User Certificate
 --------------------------
 
 This section describes how to get and set up a personal certificate to use on OSG.
-Currently you can get a certificate from either the OSG CA or from CILogon.
-After May 31st, you will only be able to get a certificate issued by CILogon.
-Other CAs may be used; if your virtual organization (VO) requires that you get a certificate from a different CA, [contact your VO Support Center](http://www.opensciencegrid.org/?pid=1000187) for instructions.
+You need an user certificate if you are going to interact directly with OSG resources or infrastructure. 
+Examples of this would be tasks like managing OASIS, directly running jobs on OSG resources, interactingly directly with 
+OSG storage elements, or updating or querying OSG site information (such as site topology).
+Currently, you can get am user certificate from CILogon.
```
Comment (brianhlin) [inline docs/security/user-certs.md]: am -> an

File: docs/security/user-certs.md [hunk 3]
```diff
@@ -10,13 +10,23 @@ Getting a User Certificate
 --------------------------
 
 This section describes how to get and set up a personal certificate to use on OSG.
-Currently you can get a certificate from either the OSG CA or from CILogon.
-After May 31st, you will only be able to get a certificate issued by CILogon.
-Other CAs may be used; if your virtual organization (VO) requires that you get a certificate from a different CA, [contact your VO Support Center](http://www.opensciencegrid.org/?pid=1000187) for instructions.
+You need an user certificate if you are going to interact directly with OSG resources or infrastructure. 
+Examples of this would be tasks like managing OASIS, directly running jobs on OSG resources, interactingly directly with 
+OSG storage elements, or updating or querying OSG site information (such as site topology).
```
Comment (brianhlin) [inline docs/security/user-certs.md]: You don't need a cert to update/query OSG site info. Only to access private contact information (e.g. email addresses/phone numbers/DNs)

File: docs/security/user-certs.md [hunk 4]
```diff
@@ -10,13 +10,23 @@ Getting a User Certificate
...
+Currently, you can get am user certificate from CILogon.
+You may also be able to user other CAs to get a certificate; if your virtual organization (VO) requires that you get a
+certificate from a different CA, [contact your VO Support Center](https://github.com/opensciencegrid/topology/tree/master/virtual-organizations) for
+instructions.
 
 ### Know your responsibilities
 
-When you request a certificate your provide some public personal information about yourself; *name, email address, phone number*, and which *VO* (virtual organization) you belong to. If any of this information changes after you have your certificate you should notify OSG (and probably VO) of the changes. For the OSG RA send email to osg-ra at opensciencegrid dot org.
+When you receive an user certificate,
```
Comment (brianhlin) [inline docs/security/user-certs.md]: Delete this line

File: docs/security/user-certs.md [hunk 5]
```diff
@@ -10,13 +10,23 @@ Getting a User Certificate
...
 
 ### Know your responsibilities
 
-When you request a certificate your provide some public personal information about yourself; *name, email address, phone number*, and which *VO* (virtual organization) you belong to. If any of this information changes after you have your certificate you should notify OSG (and probably VO) of the changes. For the OSG RA send email to osg-ra at opensciencegrid dot org.
+When you receive an user certificate, 
+When you request a certificate, you provide some public personal information about yourself; *name, email address, phone
+number*, and which *VO* (virtual organization) you belong to. 
+If any of this information changes after you have your certificate you should notify CILogon (and probably VO) of the changes. 
+In addition, if your account or user certificate is compromised, you **must** notify the issuer of your certificate. 
+For the CILogon RA send email to ca at ciloogon dot org.
```
Comment (brianhlin) [inline docs/security/user-certs.md]: This can just be `<mailto:ca@cilogon.org>`

File: docs/security/user-certs.md [hunk 6]
```diff
@@ -10,13 +10,23 @@ Getting a User Certificate
...
+certificate from a different CA, [contact your VO Support Center](https://github.com/opensciencegrid/topology/tree/master/virtual-organizations) for
+instructions.
 
 ### Know your responsibilities
 
-When you request a certificate your provide some public personal information about yourself; *name, email address, phone number*, and which *VO* (virtual organization) you belong to. If any of this information changes after you have your certificate you should notify OSG (and probably VO) of the changes. For the OSG RA send email to osg-ra at opensciencegrid dot org.
+When you receive an user certificate, 
+When you request a certificate, you provide some public personal information about yourself; *name, email address, phone
+number*, and which *VO* (virtual organization) you belong to. 
+If any of this information changes after you have your certificate you should notify CILogon (and probably VO) of the changes.
```
Comment (brianhlin) [inline docs/security/user-certs.md]: You don't need to provide any of the above in the new cilogon world order. The subsequent sentence can also be removed

File: docs/security/user-certs.md [hunk 7]
```diff
@@ -25,36 +35,23 @@ Your **name** and **email** address are encoded into the signed certificate and
 You will have to obtain your user certificate using the [CILogon web UI](https://cilogon.org/).
 Follow the steps below to get an user certificate:
 
-1. First, either search for your institution and select it or scroll through list and do the same. <br>![Institution Selection](/img/cilogon_select_idp.png)
+1. First, either search for your institution and select it or scroll through list and do the same. If your institution is not on the list, please contact your institution's IT support to see if they can support CILogon. <br>![Institution Selection](/img/cilogon_select_idp.png).
```
Comment (brianhlin) [inline docs/security/user-certs.md]: There's some extra trailing space at the nd of the sentence. Could you also put the sentence on a newline?

File: docs/security/user-certs.md [hunk 8]
```diff
@@ -25,36 +35,23 @@ Your **name** and **email** address are encoded into the signed certificate and
...
+Your certificate and key pair can be stored as separate files, by default named `~/.globus/usercert.pem` and `~/.globus/userkey.pem`, or they can be bundled in a single file in PKCS12 (Public Key Cryptography Standard \#12), by default named `~/.globus/usercred.p12`. All OSG user tools works with both formats. Unless you specify a name in the command, they look first for the certificate/key pair default names, then for the PKCS12 default name and use the first one they find. PKCS 12 is a single file, convenient to move and used by many applications, e.g. web browsers (so you don't need to convert user certificate format).
```
Comment (brianhlin) [inline docs/security/user-certs.md]: - Let's change PKCS12 into a link to the wikipedia page https://en.wikipedia.org/wiki/PKCS_12. Then get rid of the parenthetical.
- I don't think it's true that all OSG user tools work with both formats. 
- The last sentence is repetitive

File: docs/security/user-certs.md [hunk 9]
```diff
@@ -10,13 +10,23 @@ Getting a User Certificate
 --------------------------
 
 This section describes how to get and set up a personal certificate to use on OSG.
-Currently you can get a certificate from either the OSG CA or from CILogon.
-After May 31st, you will only be able to get a certificate issued by CILogon.
-Other CAs may be used; if your virtual organization (VO) requires that you get a certificate from a different CA, [contact your VO Support Center](http://www.opensciencegrid.org/?pid=1000187) for instructions.
+You need an user certificate if you are going to interact directly with OSG resources or infrastructure. 
+Examples of this would be tasks like managing OASIS, directly running jobs on OSG resources, interactingly directly with
```
[the remark this answers is not in the document]
Comment (sthapa) [reply docs/security/user-certs.md]: fixed

File: docs/security/user-certs.md (hunk 1 above)
Comment (sthapa) [reply docs/security/user-certs.md]: fixed

File: docs/security/user-certs.md (hunk 2 above)
Comment (sthapa) [reply docs/security/user-certs.md]: fixed

File: docs/security/user-certs.md (hunk 4 above)
Comment (sthapa) [reply docs/security/user-certs.md]: done

File: docs/security/user-certs.md (hunk 6 above)
Comment (sthapa) [reply docs/security/user-certs.md]: done

File: docs/security/user-certs.md (hunk 8 above)
Comment (sthapa) [reply docs/security/user-certs.md]: Done

File: docs/security/user-certs.md (hunk 5 above)
Comment (sthapa) [reply docs/security/user-certs.md]: updated

File: docs/security/user-certs.md (hunk 7 above)
Comment (sthapa) [reply docs/security/user-certs.md]: done

File: docs/security/user-certs.md (hunk 3 above)
Comment (sthapa) [reply docs/security/user-certs.md]: updated

**Comment by brianhlin** [comment]
@sthapa I think you missed this comment: 

> In addition to the comments, the PKCS12 section just needs to be redone entirely. Replace it with a section on what PKCS12 is and step-by-step instructions on how to extract the cert and key.

**Comment by sthapa** [comment]
Updated the PKCS12 section, dropped the part about troubleshooting the PKCS12 ca chain expiration but I can add a rewritten version of that in if needed.

**Comment by sthapa** [comment]
@brianhlin  mind looking at my updates?

File: docs/security/user-certs.md [hunk 10]
```diff
@@ -44,74 +44,34 @@ Follow the steps below to get an user certificate:
...
-
-None of the contacted servers for osg were capable
-of returning a valid AC for the user.
-```
 
-To solve the problem split the bundle in the two PEM files as documented in the [referenced document](http://security.ncsa.illinois.edu/research/grid-howtos/usefulopenssl.html), so that the embedded CA chain is not used.
+### Certificate formats
+
+Your user certficate can be stored in a few different formats.
+The two most common formats used on OSG are the [PKCS12](https://en.wikipedia.org/wiki/PKCS_12) and
```
Comment (brianhlin) [inline docs/security/user-certs.md]: "on the OSG" -> "in the OSG"

File: docs/security/user-certs.md [hunk 11]
```diff
@@ -44,74 +44,34 @@ Follow the steps below to get an user certificate:
...
+[PEM](https://en.wikipedia.org/wiki/Privacy-Enhanced_Mail) formats.
+In the PEM format, your user certificate is stored in two separate files: one for the certificate and another for the
+private key.
+The PKCS12 format stores the certificate and private key in a single file along with an optional certificate chain.
+Most OSG user tools will work with both but will try to use PEM files first.   
+
+To convert a PKCS12 file to  PEM files, do the following.  
+   
+1. First, extract your user certificate from your PKCS12 file.  We assume that the PKCS12 file is called `usercred.p12` and output the PEM certificate to `usercert.pem`.
```
Comment (brianhlin) [inline docs/security/user-certs.md]: Let's break up the sentences to separate lines and also add something about entering the password specified when the cert was requested

File: docs/security/user-certs.md [hunk 12]
```diff
@@ -44,74 +44,34 @@ Follow the steps below to get an user certificate:
...
+
+        :::console
+        user@host $ openssl pkcs12 -in usercred.p12 -out usercert.pem -nodes -clcerts -nokeys
+        Enter Import Password:
+        MAC verified OK
+   
+1. Second, extract the private key. The private key will be saved to userkey.pem
```
Comment (brianhlin) [inline docs/security/user-certs.md]: Let's mention and differentiate between the two different password prompts

File: docs/security/user-certs.md (hunk 10 above)
Comment (sthapa) [reply docs/security/user-certs.md]: fixed

File: docs/security/user-certs.md (hunk 11 above)
Comment (sthapa) [reply docs/security/user-certs.md]: done

File: docs/security/user-certs.md (hunk 12 above)
Comment (sthapa) [reply docs/security/user-certs.md]: done

**Comment by brianhlin** [comment]
LGTM!

**Pull request merged**